src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -242,9 +242,7 @@ public final synchronized void shutdown() throws InterruptedException
          LOGGER.info("{} - is closing down.", poolName);
          logPoolState("Before closing ");
 
-         connectionBag.close();
-         softEvictConnections();
-         addConnectionExecutor.shutdownNow();
+         addConnectionExecutor.shutdown();
```

Comment:
That is certainly not a safe bet.  What happens if Connections are taking several seconds to create each?